submission/programs/adae.R

```diff
@@ -0,0 +1,283 @@
+# ----------------------------------------------------------------------------#
+
+# ---------- #
+# read in AE #
+# ---------- #
+ae <- haven::read_xpt(file.path("sdtm", "ae.xpt"))
+suppae <- haven::read_xpt(file.path("sdtm", "suppae.xpt"))
+
+# ------------ #
+# read in ADSL #
+# ------------ #
+adsl <- haven::read_xpt(file.path("submission", "datasets", "adsl.xpt"))
+
+#----------------------------------------------------------------------------------------#
+# When SAS datasets are imported into R using haven::read_sas(), missing                 #
+# character values from SAS appear as "" characters in R, instead of appearing           #
+# as NA values. Further details can be obtained via the following link:                  #
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values #
+#----------------------------------------------------------------------------------------#
+ae <- convert_blanks_to_na(ae)
+adsl <- convert_blanks_to_na(adsl)
+
+#---------------------- #
+# ADAE derivation start #
+#---------------------- #
+
+#------------------------------------#
+# Read in specifications from define #
+#------------------------------------#
+## placeholder for origin=predecessor, use metatool::build_from_derived()
+metacore <- spec_to_metacore("adam/TDF_ADaM - Pilot 3 Team updated.xlsx", where_sep_sheet = FALSE, quiet = T)
+adae_spec <- metacore %>% select_dataset("ADAE") # Get the specifications for the dataset we are currently building
+
+# ----------------------#
+# Get list of ADSL vars #
+# ----------------------#
+adsl_vars <- vars(
+  TRTSDT,
+  TRTEDT,
+  STUDYID,
+  SITEID,
+  TRT01A,
+  TRT01AN,
+  AGE,
+  AGEGR1,
+  AGEGR1N,
+  RACE,
+  RACEN,
+  SEX,
+  SAFFL,
+  TRTSDT,
+  TRTEDT
+)
+
+#------------------#
+# Merge adsl to ae #
+#------------------#
+adae0 <- ae %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by = vars(STUDYID, USUBJID)
+  ) %>%
+  #------------------------------#
+  # Set TRTA and TRTAN from ADSL #
+  #------------------------------#
+  rename(
+    TRTA = TRT01A,
+    TRTAN = TRT01AN
+  ) %>%
+  # -----------------------------#
+  # Derive analysis start time --#
+  # -----------------------------#
+  derive_vars_dtm(
+    dtc = AESTDTC,
+    new_vars_prefix = "AST",
+    highest_imputation = "D"
+  ) %>%
+  # -----------------------------#
+  # Derive analysis end time     #
+  # -----------------------------#
+  derive_vars_dtm(
+    dtc = AEENDTC,
+    new_vars_prefix = "AEN",
+    highest_imputation = "M",
+    date_imputation = "last",
+    time_imputation = "last",
+    # max_dates = vars(DTHDT, EOSDT)
+    max_dates = NULL
+  ) %>%
+  # -----------------------------------#
+  # Derive analysis start & end dates  #
+  # -----------------------------------#
+  derive_vars_dtm_to_dt(vars(ASTDTM, AENDTM)) %>%
+  mutate(
+    TRTSDTM = as_datetime(TRTSDT),
+    astdt_m = as_datetime(as.Date(AESTDTC)),
```

Comment:
?? don't we have this information not already as astdt and aendt from previous step? This needs more explanation.

Reply:
There is an issue with the original CDISC pilot dataset. ADURN is blank for the original CDISC dataset : 
![image](https://user-images.githubusercontent.com/95468392/218565089-fae213a0-8286-4e9e-9f4c-7b2a43e4a5ae.png)
Because it seems the original SDTM.AE.AESTDTC was missing Day, where it seems the original ADAE derivation for ADURN was probably using this date instead of the imputed date. Because day is missing in AESTDTC, ADURN can't derive day.
![image](https://user-images.githubusercontent.com/95468392/218567184-771e3dcb-89e9-49f1-be72-5db6534848e1.png)

but the same records, derived in the Pilot 3 dataset do show a calculation since we are using the imputed ASTDT
![image](https://user-images.githubusercontent.com/95468392/218565707-636dc6ec-b940-4ba3-ac41-5baf4f2e1bf0.png)

So in order to match the original ADAE, we also used AESTDTC.  But we should probably go with the latter and note this in our QC findings page.

Reply:
these differences now reflected in : https://github.com/RConsortium/submissions-pilot3-adam/wiki/QC-Findings 
code is pushed to `2_dev_adae_laxamanaj`.  Please repull.

Reply:
@SHAESEN2 or @thomas-neitmann If ok, please `Resolve conversation`.